pytorch_lightning/plugins/training_type/dp.py

```diff
@@ -66,6 +66,18 @@ def setup(self) -> None:
         self.model_to_device()
         self._model = self._setup_model(LightningParallelModule(self._model))
 
+    def batch_to_device(self, batch: Any, device: Optional[torch.device] = None, dataloader_idx: int = 0) -> Any:
```

Comment:
@four4fish why did you create a `batch_to_device` method in dp.py?

Reply:
This reproduced accelerator.py L160-L164. Did you see any issue related to this change @daniellepintz?

Reply:
Right, but why both in dp.py and strategy.py? No, no issues

Reply:
Oh I see! thanks. No, no issues

Reply:
this is a bug, no? we don't need to move batch to any device in case of DP since it's handled by DataParallel already. Right now what's happening is:

```
batch - > batch_gpu_0 -> DataParallel -> batch_1_gpu_0 + batch_1_gpu_1
```
how it should be:
```
batch -> DataParallel -> batch_1_gpu_0 + batch_1_gpu_1
```
so we don't need to transfer it by ourselves.

Reply:
@rohitgr7 what's the issue here? https://github.com/PyTorchLightning/pytorch-lightning/blob/8c07d8bf905e395bbd2142b5df7b185b8e936c41/pytorch_lightning/strategies/dp.py#L73-L83

Reply:
for DP, do we need to move the batch manually to the device[0]?

Reply:
@rohitgr7 @ananthsub  I think technically DP doesn't need batch to be moved to device[0], only gradients. I kept the logic because the old accelerator.py L160-L164 has very intentionally defined `move_batch_to_device` for DP. 
I don't know why, so guessed it probably related to a bug caused by multithreading? 
I think we can test this out and remove it if it's not needed.

Reply:
Well, technically this is not required. Also tested that if we don't move it to device explictly it is working. But I saw PT example too for confirmation and there they are moving batch to device[0]. If there is a reason why this is required, then I'd love to know why else it could be updated.

Reply:
I did a quick fix some time ago https://github.com/PyTorchLightning/pytorch-lightning/pull/6098

But looks like missed a test.